DOC-2090: Document options for setting the inline dialog position (#2802)
tinymce/tinymce-docs: modules/ROOT/pages/dialog-configuration.adoc

Any callback function that is not passed a `+details+` object assumes that the dialog will only contain one component which can trigger it or that it does not matter if the function is triggered by multiple components. For example, `+onSubmit+` is only triggered when a user clicks on a *Submit* type footer button, and {productname} assumes that a dialog will only have one *Submit* type button. In comparison, `+onCancel+` and `+onClose+` are both triggered by clicking the `+X+` button in the top right of a dialog or by clicking a *Cancel* type footer button. These two buttons have the same functionality, and therefore {productname} does not differentiate between them.

== Dialog position

By default a dialog is shown as a modal in the center of the editor viewport.

This presented position can be changed by providing a second argument, with an appropriate value, to `editor.windowManager.open()`.

[[inline]]
=== `+inline+`

> In a dialog configuration, the `+inline+` option sets the dialog’s presented position to one of two possiblities.

Reviewer: The configuration for inline dialog is passed as a second argument to the `windowManager.open`, it's not part of the `dialog` configuration, so should be in the second argument. Maybe:

```suggestion
The `+inline+` option passed to the `editor.windowManager.open` sets the dialog’s presented position to one of two possiblities.
```

Each of the event callback functions - `+onAction+`, `+onSubmit+`, `+onCancel+`, `+onChange+`, `+onClose+`, and `+onTabChange+` - are shared between all dialog components that may trigger them. For example, *Custom* type footer buttons and dialog panel buttons all trigger `+onAction+`. Therefore, callback functions that may be triggered by multiple components are passed an object (called `+details+` above) that contains the `+name+` of the component that triggered the event.

Any callback function that is not passed a `+details+` object assumes that the dialog will only contain one component which can trigger it or that it does not matter if the function is triggered by multiple components. For example, `+onSubmit+` is only triggered when a user clicks on a *Submit* type footer button, and {productname} assumes that a dialog will only have one *Submit* type button. In comparison, `+onCancel+` and `+onClose+` are both triggered by clicking the `+X+` button in the top right of a dialog or by clicking a *Cancel* type footer button. These two buttons have the same functionality, and therefore {productname} does not differentiate between them.

== Dialog position

> By default a dialog presents as a modal object in the center of the editor viewport.

Reviewer 2: ```suggestion
By default a dialog is shown as a modal in the center of the editor viewport.
```

> This presented position can be changed for a given dialog by adding an `inline` value to the dialog’s configuration.

Reviewer 2: Actually, `{inline: 'bottom'}` should be provided in the second argument to windowManager.open() as seen in the example below

Author: @tiny-ben-tran,

> Actually, `{inline: 'bottom'}` should be provided in the second argument to windowManager.open() as seen in the example below

¿que?

If, by _example below_, you mean the example I added, said example doesn’t have an `{inline: 'bottom'}` value in it at all.

What am I missing?

Reviewer 2: Sorry. I meant `inline: 'toolbar'`.

Author: Appropriate edits done.

[[inline]]
=== `+inline+`

In a dialog configuration, the `+inline+` option sets the dialog’s presented position to one of two possiblities.

When added to a configuration with the value, `+toolbar+`, it sets the dialog to appear adjacent to the {productname} toolbar.

When added to a configuration with the value, `+bottom+`, it sets the dialog to appear at the bottom of the editor viewport.

*Type:* `+String+`

*Possible values:* `+toolbar+`, `+bottom+`

=== Basic example with the inline value added and set to toolbar

[source,js]
----
tinymce.activeEditor.windowManager.open({
  title: 'Dialog Title', // The dialog's title - displayed in the dialog header
  body: {
    type: 'panel', // The root body type - a Panel or TabPanel
    items: [ // A list of panel components
      {
        type: 'htmlpanel', // an HTML panel component
        html: 'Panel content goes here.'
      }
    ]
  },

>   buttons: [ // A list of footer buttons
>     {
>       type: 'submit',
>       text: 'OK'
>     }
>   ]
>   { inline: 'toolbar' }
> });

Reviewer 3: ```suggestion
  buttons: [ // A list of footer buttons
    {
      type: 'submit',
      text: 'OK'
    }
  ]
}, { inline: 'toolbar' });
```
